src/node_crypto.cc

```diff
@@ -3530,6 +3535,9 @@ bool Hash::HashInit(const char* hash_type) {
     return false;
   EVP_MD_CTX_init(&mdctx_);
   EVP_DigestInit_ex(&mdctx_, md_, nullptr);
+  if (0 != ERR_peek_error()) {
+    return false;
+  }
```

Comment:
@stefanmb Checking error stack instead of function return status is wrong as it will generate false positive in case there is previously (intentionally) ignored error. It [broke](http://bugs.alpinelinux.org/issues/4999) node on Alpine Linux.

+1 for fixing this properly.

(yeah, openssl api is funky...)